stream.go

```diff
@@ -0,0 +1,261 @@
+package psrpc
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/livekit/psrpc/internal"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type Stream[SendType, RecvType proto.Message] interface {
+	Channel() <-chan *Response[RecvType]
+	Send(SendType) error
+	Close(error) error
+	Done() <-chan struct{}
+	Err() error
```

Comment:
i'm not totally happy with this interface. the other way i considered implementing stream close was returning an error response containing the close reason from the channel. thoughts?